### PR TITLE
Switched `xarray.unfuncs` to `numpy`

### PR DESCRIPTION
**Pull request by bertilhatt**
_Doing this as part of my Hacktoberfest contribution_

## Why ## 

Trying to address Issue https://github.com/pytroll/satpy/issues/466

## How ## 

The error message clearly recommends a path to stability: switch the same functions to the `numpy` equivalents.

## Comments ## 

The only other reference to `xarray` are `xr.DataArray`. I did not convert them to `numpy` equivalent because I didn’t mention that those would be deprecated (and that would probably cascade a _lot_ more unwanted effects.

 - [ ] Closes #xxxx 
 - [ ] Tests added 
 - [ ] Tests passed 
 - [ ] Passes ``git diff origin/master -- "*py" | flake8 --diff`` 
 - [ ] Fully documented

### CODE REVIEW

**Comment by djhoese** [comment]
Just curious, did you find this specifically because of the hacktoberfest label? I haven't been using it, but apparently I should start added it.

Thanks for the PR. I'll review it soon.

File: satpy/composites/viirs.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 import dask
 import dask.array as da
 import xarray as xr
-import xarray.ufuncs as xu
+# import xarray.ufuncs as xu
```
Comment (djhoese) [inline satpy/composites/viirs.py]: You can just remove this if it isn't being used.

**Comment by djhoese** [comment]
@bertilhatt do you plan on continuing work on this pull request and fixing more things? If not I feel I have to mark this as invalid/spam. You did the bare minimum to get your hacktoberfest PR and are just making more work for me by doing so.

**Comment by bertilhatt** [comment]
Sorry, I’ve been trying to reproduce the PendingDeprecationWarning but all I see are error messages because of other files missing.
``` 
Running pykdtree-1.3.1/setup.py -q bdist_egg --dist-dir /var/folders/w3/43yvkc5s3kn675sdt8lg1_qm0000gn/T/easy_install-nfh5g9ec/pykdtree-1.3.1/egg-dist-tmp-0n0f4wb6
warning: no previously-included files found matching 'pykdtree/render_template.py'
clang: error: unsupported option '-fopenmp'
error: Setup script exited with error: command 'clang' failed with exit status 1
```
I understand it’s a dependency on a C package, I was hoping to look into it last night; I haven’t had the time to dig into what my local setup got wrong.

**Comment by bertilhatt** [comment]
With OMP off for `pykdtree`, I am seeing now seeing warning but they seem to be problems in C:
```
…
Running pyproj-1.9.5.1/setup.py -q bdist_egg --dist-dir /var/folders/w3/43yvkc5s3kn675sdt8lg1_qm0000gn/T/easy_install-a9k7nfj6/pyproj-1.9.5.1/egg-dist-tmp-227e_az9
using bundled proj4..
nad2bin.c: In function 'main':
nad2bin.c:362:21: warning: comparison of integer expressions of different signedness: 'size_t' {aka 'long unsigned int'} and 'int' [-Wsign-compare]
                     != 4 * ct.lim.lam )
                     ^~
…
```
Not sure those are tied to this project.

Do you have a version of the Python Warnings that you are seeing?

**Comment by djhoese** [comment]
Try installing the satpy dependencies before running `python setup.py test`. Normally I use conda with the conda-forge channel to install satpy, but everything should work with pip/PyPI. Did the tests run even with the C warnings from above? Try `pip install pyresample pyproj trollimage pyorbital pyspectral trollsift` and see if that installs fine. Then you can try the tests again.

I do not have a list of all of the warnings, sorry. You'd want to run the tests again to make sure the warnings are gone anyway.

**Comment by djhoese** [comment]
Any updates on this?

**Comment by djhoese** [comment]
I wonder if this is a pyproj incompatibility with python 3.7 (I haven't run out full stack on 3.7 yet). Since pyproj isn't our project I'm not sure I can do anything. Can you do `pip install pyproj`?

**Comment by bertilhatt** [comment]
Same error:
https://gist.github.com/bertilhatt/b78bc2c5b2994ec12616792011438ff9

**Comment by djhoese** [comment]
https://github.com/jswhit/pyproj/issues/136

Pyproj doesn't support python 3.7. I guess for now you'll either need to do python 3.6 and/or use conda to ease any other installation issues.

**Comment by bertilhatt** [comment]
I was able to configure `pyenv` to use Python 3.6.6 locally and run the `pip install` that you mentioned.

Here’s by bash history:
https://gist.github.com/bertilhatt/75fcbf52411b765a56fe581c80dc2cc3

Let me know if those are the error message mentioned in #466.

**Comment by bertilhatt** [comment]
I’ve gone through the error and warning message that I can see. All of those see to be related to C++ types, which is something that I’m not enough familiar with to help with.

**Comment by djhoese** [comment]
Those errors are not from our tests. Those are still installing the dependencies for satpy which still seem to be failing for you. Sorry this has been so difficult. Due to an issue with the naming of the `pyhdf` package your installation is failing (`pip install python-hdf4` versus conda's `pip install pyhdf`). It may be easiest to just run `python -m unittest satpy.tests` from the root of the repository. You'll get certain import errors and can install anything that is missing.

**Pull request merged**